demo/site/src/common/blocks/InternalLinkBlock.tsx

```diff
@@ -14,16 +15,23 @@ export function InternalLinkBlock({ data: { targetPage, targetPageAnchor }, chil
         return <span className={className}>{children}</span>;
     }
 
-    let href = targetPageAnchor !== undefined ? `${targetPage.path}#${targetPageAnchor}` : targetPage.path;
-    if (targetPage.scope) {
-        const language = (targetPage.scope as Record<string, string>).language;
-        if (language) {
-            href = `/${language}${href}`;
-        }
+    if (targetPage.scope == null) {
+        throw new Error("InternalLinkBlock: targetPage.scope is required");
     }
 
+    const pathWithAnchor = targetPageAnchor !== undefined ? `${targetPage.path}#${targetPageAnchor}` : targetPage.path;
+
     return (
-        <Link href={href} title={title} className={className}>
+        <Link
+            href={createSitePath({
+                scope: {
+                    language: (targetPage.scope as Record<string, string>).language,
+                },
```

Comment:
IMO this should be possible to use `scope: targetPage.scope`. If this doesn't work, then it should be `scope: targetPage as PageTreeNodeScope`

Reply:
targetPage.scope is typed as unkown

<img width="519" alt="Screenshot 2025-03-06 at 16 04 35" src="https://github.com/user-attachments/assets/3c127565-5d32-4f57-acea-98b0cce8a69d" />

Have done a hard cast with ` as GQLPageTreeNodeScope`.